meeting-notes/2018/2018-08-13--golang-core-dev-team-weekly.md

```diff
@@ -0,0 +1,184 @@
+# 💫Golang Core Dev Team Weekly Sync 🙌🏽 August 13, 2018
+
+- **Lead:** David Dias
+- **Notetaker:** @keks
+- **Attendees:**
+  - @stebalien
+  - @diasdavid
+  - @bigs
+  - @Magik6k
+  - @keks
+  - @djdv
+  - @hsanjuan
+  - @johnnycrunch
+  
+- **Recording:** Recording is currently private to the Go Core Dev Team.
+
+## Agenda
+
+- Ask everyone to put their name into the list of attendees
+- Round of updates
+  - What have you accomplished since the last Weekly?
+  - Were there any blockers? If so, which ones? Is it still blocked? Why?
+  - What is the next important thing you should focus on?
+- Ask for general questions. Could be things like:
+  - I'm stuck with something, I don't know who to ask. Who knows who to ask?
+  - Who can help me with xyz?
+- Plan this week
+  - Select issues to work on
+- Review remaining issues if there is time left
+
+
+## Notes
+
+### Week Update
+
+@stebalien
+ - Done:
+   - Reviewed a bunch of PRs
+   - Responded to/triaged a bunch of issues
+   - Fixed a few bugs
+   - Notable: update gogo protobuf
+ - Blocked:
+   - https://github.com/multiformats/multiaddr/pull/68 <- @daviddias P3
+   - https://github.com/multiformats/multiaddr/pull/68#issuecomment-401971665
+ - Next:
```

Comment:
@Stebalien also, as discussed during the call. Please do the drawing (even if on paper) of something like:
https://github.com/ipfs/js-ipfs#code-architecture-and-folder-structure
![image](https://user-images.githubusercontent.com/1211152/44144035-4d7c5e62-a086-11e8-8607-1098a75b31c6.png)

But for go. Contemplating what is legacy, what is next, including:

- go-ipfs-cli
- go-ipfs-cmds
- go-ipfs-api
- go-ipfs-core
- go-ipfs daemon
  - go-ipfs-http-api
  - go-ipfs-http-gateway

And anything else that it is there.

Reply:
Bonus points if you do another for how things are pieced together. Like this one https://github.com/ipfs/js-ipfs#ipfs-core-architecture

Reply:
**You can do it on paper**, I can create the ascii version

Reply:
Diagram:

![commands](https://user-images.githubusercontent.com/310393/44173527-bc24c180-a0cf-11e8-8729-991a8b57dd0a.png)

Source: https://gist.github.com/Stebalien/798f1d8d672f365ef1030fc810cea3ea

Reply:
Dotted means "likely going away". The "Legacy" parts are thin wrappers around some commands to translate between the new system and the old system. The grayed-out parts on the "daemon" diagram are there to show that the code is all the same, it's just that we turn some pieces on and some pieces off depending on whether we're running on the client or the server.

Reply:
Added to the README with https://github.com/ipfs/go-ipfs/pull/5396